Fix one bug get one for free?
```
=================================== FAILURES ===================================
________________________ ToolsTestCase.test_search_grep ________________________

self = <galaxy_test.api.test_tools.ToolsTestCase testMethod=test_search_grep>

    @skip_without_tool("Grep1")
    def test_search_grep(self):
        url = self._api_url("tools")
        payload = dict(q="Select lines that match an expression")
        get_response = get(url, payload).json()
>       assert "Grep1" in get_response
E       AssertionError: assert 'Grep1' in ['Show beginning1', 'Show tail1', 'head', 'random_lines1', 'Filter1', 'Summary_Statistics1', ...]

lib/galaxy_test/api/test_tools.py:131: AssertionError
----------------------------- Captured stdout call -----------------------------
INFO:     127.0.0.1:57360 - "GET /api/users HTTP/1.1" 200 OK
INFO:     127.0.0.1:57362 - "POST /api/users/adb5f5c93f827949/api_key HTTP/1.1" 200 OK
INFO:     127.0.0.1:57364 - "GET /api/tools?in_panel=False HTTP/1.1" 200 OK
INFO:     127.0.0.1:57366 - "GET /api/tools?q=Select+lines+that+match+an+expression HTTP/1.1" 200 OK
INFO:     127.0.0.1:57368 - "GET /api/jobs?state=running&?user_details=true HTTP/1.1" 200 OK
```

--- a/lib/galaxy/tools/search/__init__.py
+++ b/lib/galaxy/tools/search/__init__.py
@@ -101,7 +101,7 @@ class ToolBoxSearch:
             for tool_id in tool_ids_to_remove:
                 writer.delete_by_term('id', tool_id)
             for tool_id in tool_cache._new_tool_ids - indexed_tool_ids:
-                tool = tool_cache.get_tool_by_id(tool_id)
+                tool = self.toolbox.get_tool(tool_id)
                 if tool and tool.is_latest_version:
                     if tool.hidden:
                         # we check if there is an older tool we can return

